torch/_inductor/lowering.py

```diff
@@ -3358,6 +3358,16 @@ def _unsafe_index_put_(self, indices, values, accumulate=False):
 
 
 def index_put_impl_(self, indices, values, accumulate, check):
+    def try_get_name(x):
+        if isinstance(x, ir.TensorBox):
+            x = x.data
+        if isinstance(x, ir.StorageBox):
+            x = x.data
+        return x.get_name() if isinstance(x, ir.Buffer) else None
```

Comment:
Is it possible that `x` is a view which also has name?

Reply:
Thanks for the comment, adding the handling code for view like: `TensorBox -> View -> StorageBox -> Buffer`